Automattic/wp-calypso
# Full Site Editing: add beta enrollment step to New Onboarding flow

#### Changes proposed in this Pull Request

* add a beta enrollment step to the `/new` flow

**NOTE**: this still does not include a strategy (user mod, probably) for selectively revealing the beta step.

#### Testing instructions

1. Navigate to `calypso.localhost:3000/new`
2. You should see `beta` as the 2nd step if you skip the 1st `IntentGathering` ("My Site is Called ___") step, or the 3rd step if you enter something there (`domains` comes 2nd in that case)
3. On the theme picker (`design`)  step, you should see only FSE themes if you've enrolled, and no FSE themes if you haven't
4. If enrolled in beta, you **should not** see the fonts (`style`) step
5. Ensure that "Go Back" and "Skip for now" links work predictably throughout.
6. Ensure that your site shows the Site Editor on the created site if you opted in, and vice versa
7. Try different viewport sizes

Large screen:
<img width="1067" alt="Screen Shot 2021-09-23 at 14 13 03" src="https://user-images.githubusercontent.com/195089/134569356-ec79fe89-ba13-412c-96cf-0be5c162cb93.png">

Small screen:
<img width="375" alt="Screen Shot 2021-09-23 at 14 15 35" src="https://user-images.githubusercontent.com/195089/134569654-25a2c3af-099b-47db-8bf2-75ce57134c82.png">


Supersedes #56175
Closes #54461
